### PR TITLE
Implements Database

### PR DESCRIPTION
**Pull request by gui1612**

### CODE REVIEW

File: README.md
Comment (Naapperas) [inline README.md:1]: Now that the repository name and purpose changed, you need to change this README to better reflect your changes.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      POSTGRES_DB: ${POSTGRES_DB}
+    ports:
+      - "5432:5432"
+    networks:
+      - timescale-network
+    volumes:
+      - ./init-scripts/schema.psql:/docker-entrypoint-initdb.d/init.sql
+
+networks:
+  timescale-network:
```
Comment (Naapperas) [inline docker-compose.yml:18]: Why do you have a network just for timescale? Docker Compose already creates a default network for all the services defined here. Couldn't that be used?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    environment:
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_DB: ${POSTGRES_DB}
+    ports:
+      - "5432:5432"
+    networks:
+      - timescale-network
+    volumes:
+      - ./init-scripts/schema.psql:/docker-entrypoint-initdb.d/init.sql
```
Comment (Naapperas) [inline docker-compose.yml:15]: I understand the use of the volume directive here, but perhaps we could build our own image (which is based on timescale's one) and just do our modifications there. This is not a "change request" so much as it is a "discussion request". Food for thought.

File: init-scripts/schema.psql [hunk 3]
```diff
@@ -0,0 +1,36 @@
+-- drop old schema
+DROP TABLE IF EXISTS DeviceData;
+DROP TABLE IF EXISTS Device;
+
+-- device table
+CREATE TABLE Device(
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    model VARCHAR(255),
+    serial_number VARCHAR(255),
```
Comment (Naapperas) [inline init-scripts/schema.psql:10]: Why are these nullable while the others are not?

File: init-scripts/schema.psql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    
+INSERT INTO DeviceData (device_id, value) VALUES
+    (1, '{"battery_level": 80, "screen_brightness": 70, "location": {"latitude": 37.7749, "longitude": -122.4194}}'),
+    (1, '{"battery_level": 75, "screen_brightness": 60, "location": {"latitude": 34.0522, "longitude": -118.2437}}'),
+    (2, '{"battery_level": 95, "cpu_usage": 25, "ram_usage": 60, "storage_capacity": 512}'),
+    (2, '{"battery_level": 90, "cpu_usage": 30, "ram_usage": 55, "storage_capacity": 256}'),
+    (3, '{"temperature": 72, "humidity": 40, "mode": "cooling"}'),
+    (3, '{"temperature": 68, "humidity": 45, "mode": "heating"}'),
+    (4, '{"heart_rate": 78, "steps": 5000, "sleep_duration": 7}'),
+    (4, '{"heart_rate": 82, "steps": 6000, "sleep_duration": 8}');
```
Comment (Naapperas) [inline init-scripts/schema.psql:36]: This should not be here: this file should only be concerned with schema definitions. We can create mock data but it should be stored elsewhere.

File: init-scripts/schema.psql
Comment (Naapperas) [inline init-scripts/schema.psql:1]: Perhaps you should write comments on why you chose these specific datatypes with these specific constraints.

**Review by Naapperas — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -42,7 +45,6 @@ Note: If you need the username/password to run the database, you should contact
 
 ### Initializing the Database
 
-Before using the database module, you need to initialize the database by creating tables and necessary schema.
-
 ```bash
-docker-compose exec database-container node initialize-database.js
+docker exec -it database-timescale-1 psql -U <username> -d ThingzLinkerDB
```
Comment (ttoino) [inline README.md:49]: Using `docker compose exec` is safer here, as it doesn't rely on the container name following this convention.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,6 @@
+FROM timescale/timescaledb:latest-pg14
```
Comment (ttoino) [inline Dockerfile:1]: Shouldn't we use the latest available version of postgres (15)?

**Review by ttoino — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 6 above)
Comment (gui1612) [reply Dockerfile:1]: To be honest, I just followed [timescale's official docs](https://docs.timescale.com/self-hosted/latest/install/installation-docker/). We change it to postgres 15 if you believe there are reasons to do so

**Review by Naapperas — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (ttoino) [reply README.md]:
```suggestion
docker compose exec -it timescale psql -U <username> -d ThingzLinkerDB
```

**Review by ttoino — APPROVED** [review]

**Review by Naapperas — APPROVED** [review]

**Pull request merged**